aesara/link/numba/dispatch/tensor_basic.py

```diff
@@ -224,3 +225,17 @@ def scalar_from_tensor(x):
         return x.item()
 
     return scalar_from_tensor
+
+
+@numba_funcify.register(TransposeOp)
+def numba_funcify_TransposeOp(op, **kwargs):
+    node = kwargs["node"]
+
+    new_order_constant = node.inputs[1]
+    new_order_tuple = tuple(new_order_constant.eval())
+
+    @numba.njit(inline="always")
+    def transpose(x, new_order):
+        return x.transpose(new_order_tuple)
+
+    return transpose
```

Comment:
The Numba versions of these Ops will require https://github.com/numba/numba/pull/7604 to go in before we can have an implementation here. 

(We can temporarily use a `np.reshape` approach similar to `Dimshuffle` is we decide to replace `Dimshuffle` ASAP)